src/Microsoft.Dnx.Host.Clr/RuntimeBootstrapper.cs

```diff
@@ -9,23 +9,35 @@ internal class RuntimeBootstrapper
     {
         public static int Execute(string[] argv, FrameworkName targetFramework)
         {
-            var executeMethodInfo = GetBootstrapperType()
-                .GetMethod("Execute", BindingFlags.Static | BindingFlags.Public, null, new[] { typeof(string[]), typeof(FrameworkName) }, null);
-            return (int)executeMethodInfo.Invoke(null, new object[] { argv, targetFramework });
+            var bootstrapperContext = GetBootstrapperContext();
+
+            var executeMethodInfo = GetBootstrapperType(bootstrapperContext.GetType().Assembly)
+                .GetMethod("Execute", BindingFlags.Static | BindingFlags.Public, null,
+                    new[] { typeof(string[]), typeof(FrameworkName), bootstrapperContext.GetType() }, null);
+            return (int)executeMethodInfo.Invoke(null, new object[] { argv, targetFramework, null });
         }
 
         // This method is only called by Helios
         public static Task<int> ExecuteAsync(string[] argv, FrameworkName targetFramework)
         {
-            var executeMethodInfo = GetBootstrapperType()
-                .GetMethod("ExecuteAsync", BindingFlags.Static | BindingFlags.Public, null, new[] { typeof(string[]), typeof(FrameworkName) }, null);
-            return (Task<int>)executeMethodInfo.Invoke(null, new object[] { argv, targetFramework });
+            var bootstrapperContext = GetBootstrapperContext();
+
+            var executeMethodInfo = GetBootstrapperType(bootstrapperContext.GetType().Assembly)
+                .GetMethod("ExecuteAsync", BindingFlags.Static | BindingFlags.Public, null,
+                    new[] { typeof(string[]), typeof(FrameworkName), bootstrapperContext.GetType() }, null);
+            return (Task<int>)executeMethodInfo.Invoke(null, new object[] { argv, targetFramework, null });
         }
 
-        private static Type GetBootstrapperType()
+        private static Type GetBootstrapperType(Assembly dnxHost)
         {
-            var dnxHost = Assembly.Load("Microsoft.Dnx.Host");
             return dnxHost.GetType("Microsoft.Dnx.Host.RuntimeBootstrapper");
         }
+
+        private static object GetBootstrapperContext()
+        {
+            var dnxHost = Assembly.Load("Microsoft.Dnx.Host");
+            dynamic bootstrapperContext = Activator.CreateInstance(dnxHost.GetType("Microsoft.Dnx.Host.BootstrapperContext"));
```

Comment:
OK. I will use reflection to set the properties (for Desktop CLR we will flow the by extending ApplicationMainInfo struct https://github.com/aspnet/dnx/blob/dev/src/Microsoft.Dnx.Host.Clr/DomainManager.cs#L167